### ISSUE TITLE
the server truncates files before drop database

### ISSUE THREAD

**Issue by grossvater** [orientechnologies/orientdb#5336]
During my attempt to fast copy a database using hard links (see #5328) I found that the server (v2.1.5) truncates some files before dropping a database and I wonder if this is the intended behaviour.

The lines to be blamed seems to be inside `com.orientechnologies.orient.core.storage.cache.local.OWOWCache`:

```
  private String doDeleteFile(int fileId) throws IOException {
    if (isOpen(fileId)) {
        truncateFile(fileId);
    }
    ...
 }
```

**Comment by laa** [comment]
@grossvater I think we may merely clear write cache (which is done in truncate), but what is problem with such kind of behaviour

**Comment by laa** [comment]
Closed because of no reply. Please free to reopen issue if you have information which I asked.

**Comment by grossvater** [comment]
Sorry for my late reply. 

@laa I already explained in the description of the problem: if the files are hard linked to some other directory, I would like to preserve to files, when I drop the database.

**Comment by laa** [comment]
@grossvater sorry I do not get it. According to our API design if we drop database we remove files from disk. If you wish to preserve files from database on server merely move database directory to another location  and server will not see it. Why do you need such functionality, could you add additional explanation ?

**Comment by grossvater** [comment]
My original issue (#5328) described what I need and why I tried to hard link the files: because I need a way to detach the database while the server is running, without deleting the files.

**Comment by laa** [comment]
Ok, but we can not break server API . And I still do not understand. 
If we will remove files any way during deletion of  storage how prevention of truncation of them will help to solve issue ?

**Comment by grossvater** [comment]
Because I hard link every file to another directory (simple like this, `ln db/file1 backup/file1' for every file in database directory).

@laa Anyway, why do you say you break the server api? You actually change the implementation not the api: delete the file but do not truncate, but the file will still disappear from database directory and this will be an acceptable solution for me. 

But the perfect solution for me would be a "detach database" feature, of course.

**Comment by laa** [comment]
@grossvater well I see now what you mean, will be at next 2.2 rc version release.

**Comment by taburet** [comment]
Fixed on develop branch. Will be available in next 2.2 RC.
